chia/full_node/mempool.py

```diff
@@ -90,66 +172,76 @@ def remove_from_pool(self, items: List[bytes32], reason: MempoolRemoveReason) ->
         """
         Removes an item from the mempool.
         """
-        for spend_bundle_id in items:
-            item: Optional[MempoolItem] = self._spends.get(spend_bundle_id)
-            if item is None:
-                continue
-            assert item.name == spend_bundle_id
-            removals: List[Coin] = item.removals
-            for rem in removals:
-                rem_name: bytes32 = rem.name()
-                self._removal_coin_id_to_spendbundle_ids[rem_name].remove(spend_bundle_id)
-                if len(self._removal_coin_id_to_spendbundle_ids[rem_name]) == 0:
-                    del self._removal_coin_id_to_spendbundle_ids[rem_name]
-            del self._spends[item.name]
-            del self._sorted_spends[item.fee_per_cost][item.name]
-            dic = self._sorted_spends[item.fee_per_cost]
-            if len(dic.values()) == 0:
-                del self._sorted_spends[item.fee_per_cost]
-            self._total_mempool_cost = CLVMCost(uint64(self._total_mempool_cost - item.cost))
-            self._total_mempool_fees = self._total_mempool_fees - item.fee
-            assert self._total_mempool_cost >= 0
-            info = FeeMempoolInfo(self.mempool_info, self._total_mempool_cost, self._total_mempool_fees, datetime.now())
-            if reason != MempoolRemoveReason.BLOCK_INCLUSION:
-                self.fee_estimator.remove_mempool_item(
-                    info, MempoolItemInfo(item.cost, item.fee, item.height_added_to_mempool)
-                )
+        if items == []:
+            return
+
+        removed_items: List[MempoolItemInfo] = []
+        if reason != MempoolRemoveReason.BLOCK_INCLUSION:
+
+            for spend_bundle_ids in chunks(items, SQLITE_MAX_VARIABLE_NUMBER):
+                args = ",".join(["?"] * len(spend_bundle_ids))
+                with self._db_conn:
+                    cursor = self._db_conn.execute(
+                        f"SELECT name, cost, fee FROM tx WHERE name in ({args})", spend_bundle_ids
+                    )
+                    for row in cursor:
+                        name = bytes32(row[0])
+                        internal_item = self._items[name]
+                        item = MempoolItemInfo(int(row[1]), int(row[2]), internal_item.height_added_to_mempool)
+                        removed_items.append(item)
+
+        for name in items:
+            self._items.pop(name)
+
+        for spend_bundle_ids in chunks(items, SQLITE_MAX_VARIABLE_NUMBER):
+            args = ",".join(["?"] * len(spend_bundle_ids))
+            with self._db_conn:
+                self._db_conn.execute(f"DELETE FROM tx WHERE name in ({args})", spend_bundle_ids)
```

Comment:
What happens if we try to delete a nonexistent tx?

It's probably worth checking in tests with a malformed DB in each of these new SQL query cases, so we can test that the code calling these methods do what we expect.

Reply:
it's not an error to include names in the `in (...)` statements that don't match any rows in the table

Reply:
but we delete out of the python dict as well, which *does* fail if we try to delete an item that doesn't exist